Source/WebCore/css/DeprecatedCSSOMValue.cpp

```diff
@@ -85,15 +85,20 @@ unsigned short DeprecatedCSSOMComplexValue::cssValueType() const
     constexpr unsigned short CSS_INITIAL = 4;
     constexpr unsigned short CSS_UNSET = 5;
     constexpr unsigned short CSS_REVERT = 6;
-
-    if (m_value->isInheritValue())
-        return CSS_INHERIT;
-    if (m_value->isInitialValue())
-        return CSS_INITIAL;
-    if (m_value->isUnsetValue())
-        return CSS_UNSET;
-    if (m_value->isRevertValue())
-        return CSS_REVERT;
+    if (auto primitiveValue = dynamicDowncast<CSSPrimitiveValue>(m_value.get())) {
+        switch (primitiveValue->valueID()) {
```

Comment:
Why not just `switch(valueID(m_value.get()))`?

Reply:
Sure, good suggestion.